Simple Search Engine/task/src/search/SearchApp.java

```diff
@@ -0,0 +1,138 @@
+package search;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class SearchApp {
+    private final Scanner scanner;
+    private final SearchIndex searchIndex;
+    private boolean isUpAndRunning;
+    private ISearchMethod method;
+
+    public SearchApp() {
+        this.scanner = new Scanner(System.in);
+        this.isUpAndRunning = true;
+        this.searchIndex = new SearchIndex();
+    }
+
+    public void setMethod(ISearchMethod method) {
+        this.method = method;
+    }
+
+    public ArrayList<String> searchFor(String[] queryParts, SearchIndex searchIndex) {
+        return method.searchFor(queryParts, searchIndex);
+    }
+
+    private String[] askSearchQuery() {
+        System.out.println("Enter a name or email to search all suitable people.");
+        return scanner.nextLine().toLowerCase().split(" ");
+    }
+
+    private String askSearchMethod() {
+        System.out.println("Select a matching strategy: ALL, ANY, NONE");
+        return scanner.nextLine();
+    }
+
+    boolean isUpAndRunning() {
+        return isUpAndRunning;
+    }
+
+    void runMenuAction() {
+        final String menu = "\n=== Menu ===\n" +
+                "1. Find a person\n" +
+                "2. Print all people\n" +
+                "0. Exit";
+
+        System.out.println(menu);
+        String action = scanner.nextLine();
+
+        switch (action) {
+            case "1":
+                findAPersonByMethod();
+                break;
+            case "2":
+                printDataStore();
+                break;
+            case "0":
+                exitApp();
+                break;
+            default:
+                System.out.println("Incorrect option! Try again.");
+                break;
+        }
+    }
+
+    /**
+     * Method that does the searching, based on the chosen search algorithm, the algorithm is changed on runtime.
+     */
+    private void findAPersonByMethod() {
+        String searchMethod = askSearchMethod();
+
+        switch (searchMethod) {
+            case "ALL":
+                setMethod(new SearchMethodAll());
+                break;
+            case "ANY":
+                setMethod(new SearchMethodAny());
+                break;
+            case "NONE":
+                setMethod(new SearchMethodNone());
```

Comment:
Dit is een stap de goeie kant op, maar nog niet zoals het zou moeten. Het kan zonder setMethod. Kijk namelijk is goed wat je daarmee doet en vraag dan aan jezelf heb ik die methode nodig?

Reply:
Done.